packages/babylon/src/parser/statement.js

```diff
@@ -1368,6 +1368,9 @@ export default class StatementParser extends ExpressionParser {
       return this.parseFunction(expr, true, false, true, true);
     } else if (this.match(tt._class)) {
       return this.parseClass(expr, true, true);
+    } else if (this.match(tt.at)) {
+      this.parseDecorators(false);
+      return this.parseClass(expr, true, false, false, true);
```

Comment:
Last thing, you passed too many parameters here. It should be `expr, true, true` like the line above.